mirny.py

```diff
@@ -153,6 +153,40 @@ def connect_ext(self, ext, adr, mask):
             ),
         ]
 
+    def connect_almazny_passthrough(self, pt, adr, mask):
+        # for use w/ almazny
+        self._check_intersection(adr, mask)
+        self._slaves.append((pt, adr, mask))
+        stb = AsyncRst()
+        self.submodules += stb
+
+        saved_adr = Signal(2)
+        latch_signal = Signal()
+
+        self.comb += [
+            stb.i.eq(((self.bus.adr & adr) == adr) & (self.ext.cs)),
```

Comment:
Why does this differ from the logic in `connect_ext`?

Reply:
bus.adr stays up in unchanged form until the next SPI transaction, meaning if it was left up, at least in current form, latch signal would not rise until the next transaction - functionally meaning, if you set attenuators for one channel, they change would actually not apply until you changed another one.

That's the main problem with driving these shift registers, as they're not exactly SPI extensions. CS signal however goes down right after last bit of data is sent.

Reply:
That's exactly what is solved in `connect_ext`. Please have another look.

Reply:
That's exactly what is solved in `connect_ext`. Please have another look.

Reply:
That's exactly what is solved in `connect_ext`. Please have another look.